ydb/library/yql/minikql/comp_nodes/mkql_grace_join_imp.cpp

```diff
@@ -140,17 +127,18 @@ void TTable::AddTuple(  ui64 * intColumns, char ** stringColumns, ui32 * strings
     char ** dataStringsColumns = stringColumns + NumberOfKeyStringColumns;
     ui32 * dataStringsSizes = stringsSizes + NumberOfKeyStringColumns;
 
+    ui64 initialStringsSize = stringVals.size();
     for( ui64 i = 0; i < NumberOfDataStringColumns; i++) {
         ui32 currStringSize = *(dataStringsSizes + i);
         stringVals.insert(stringVals.end(), *(dataStringsColumns + i), *(dataStringsColumns + i) + currStringSize);
     }
 
     for ( ui64 i = 0; i < NumberOfDataIColumns; i++) {
         stringVals.insert( stringVals.end(), IColumnsVals[NumberOfKeyIColumns + i].begin(), IColumnsVals[NumberOfKeyIColumns + i].end());
-
     }
 
-
+    TableBucketsStats[bucket].KeyIntValsTotalSize += keyIntVals.size() - offset;
```

Comment:
Why do we reduce size by offset? Maybe we can just use assignment?
`TableBucketsStats[bucket].KeyIntValsTotalSize = keyIntVals.size()`

Reply:
In `TableBucketsStats[bucket].KeyIntValsTotalSize` we want to have total size of `keyIntVals` vector, because we use this size to calculate correct indexes (offsets).

After spilling the bucket `keyIntVals.size()` will be equal to 0. But we want to keep indexing like there are no spilling. Because after restoring all the vectors from spilling indexes must be correct. 

That is why here I increase the size of  `KeyIntValsTotalSize` by the number of added elements. Number of added elements is computed like this:

```
offset = keyIntVals.size();

keyIntVals.AddElements();

KeyIntValsTotalSize  += keyIntVals.size() - offset;
```

I believe it would be more clear if i used `InitialSize` instead of `offset`